File: ubuntu/pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>
#include <string.h>

int i;
int n;
int letter_count;
int sentence_count;
int word_count = 1;
float grade;

int main(void)
{
    string s = get_string("Text: ");
    
    for (i = 0, n = strlen(s); i < n; i++)
    {
        if ((s[i] >= 'A' && s[i] <= 'Z') || (s[i] >= 'a' && s[i] <= 'z'))
        { 
            letter_count += 1;   
        }
    }
    
    for (i = 0, n = strlen(s); i < n; i++)
    {
        if (s[i] == '?' || s[i] == '.' || s[i] == '!')
        {
            sentence_count += 1;
        }
    }
    
    for (i = 0, n = strlen(s); i < n; i++)
    {
        if (s[i] == ' ')
        {
            word_count += 1;
        }
    }
    
    grade = 0.0588 * (100 * (float) letter_count / (float) word_count) - 0.296 * (100 * (float) sentence_count / (float) word_count) - 15.8;
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(grade));
    }
}

File: ubuntu/pset6/readability/readability.py
from cs50 import get_string

def main():
    text = get_string("Text: ")
    
    text_length = len(text)
    
    space_count = 0
    sentences = 0
    letter_count = 0
    
    for i in range(text_length):
        if str.isalpha(text[i]) == True:
            letter_count += 1
        if str.isspace(text[i]) == True:
            space_count += 1
        if (text[i] == '.') or (text[i]=='!') or (text[i]=='?'):
            sentences += 1
    
    grade = 0.0588 * (100 * letter_count / (space_count + 1)) - 0.296 * (100 * sentences / (space_count + 1)) - 15.8
    
    if (grade < 1):
        print("Before Grade 1")
    elif (grade > 16):
        print("Grade 16+")
    else:
        print(f"Grade {round(grade)}")

main()